src/msg/async/Timeout.h

```diff
@@ -0,0 +1,44 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2024 IONOS SE
+ *
+ * Author: Max Kellermann <max.kellermann@ionos.com>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#ifndef CEPH_MSG_TIMEOUT_H
```

Comment:
Do we need a dedicated header? There is already `common/ceph_time.h`.

Reply:
Need not, but I thought this was a special function that is tailored for converting timeouts and nothing else, and is only ever used by `msg/async/Event.cc`. And `ceph_time.h` comes with heavy dependencies on `<string>`, fmt, `<ostream>` and even `<fstream>` (via `fmt/ostream.h`).
I have a passionate hate for fat headers with a deep dependency tree. And I long for the day C++20 modules solve this problem once and for all, and Ceph could switch to them any day if you want to. (cmake supports modules, but I prefer Meson over cmake, but unfortunately Meson won't support modules for a long time.)